### PR TITLE
Flux Kontext

### PR DESCRIPTION
**Pull request by a-r-r-o-w**

### CODE REVIEW

File: docs/source/en/api/pipelines/flux.md [hunk 1]
```diff
@@ -273,6 +274,46 @@ images = pipe(
...
+
+Flux Kontext is a model that allows in-context control of the image generation process, allowing for editing, refinement, relighting, style transfer, character customization, and more.
+
+```python
+import torch
+from diffusers import FluxKontextPipeline
+from diffusers.utils import load_image
+
+pipe = FluxKontextPipeline.from_pretrained(
+    "black-forest-labs/FLUX.1-kontext", transformer=transformer, torch_dtype=torch.bfloat16
```
Comment (yiyixuxu) [inline docs/source/en/api/pipelines/flux.md]: are we missing `transformer = FluxTransformer2DModel.from_pretrained(...)` ?

File: src/diffusers/pipelines/flux/pipeline_flux_kontext.py [hunk 2]
```diff
@@ -0,0 +1,1117 @@
...
+        # Flux latents are turned into 2x2 patches and packed. This means the latent width and height has to be divisible
+        # by the patch size. So the vae scale factor is multiplied by the patch size to account for this
+        self.latent_channels = self.vae.config.latent_channels if getattr(self, "vae", None) else 16
+        self.image_processor = VaeImageProcessor(vae_scale_factor=self.vae_scale_factor * 2)
+        self.tokenizer_max_length = (
+            self.tokenizer.model_max_length if hasattr(self, "tokenizer") and self.tokenizer is not None else 77
+        )
+        self.default_sample_size = 128
+
+    def _get_t5_prompt_embeds(
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/flux/pipeline_flux_kontext.py:261]: `# Copied from?`

File: src/diffusers/pipelines/flux/pipeline_flux_kontext.py [hunk 3]
```diff
@@ -0,0 +1,1116 @@
...
+
+        _, seq_len, _ = prompt_embeds.shape
+
+        # duplicate text embeddings and attention mask for each generation per prompt, using mps friendly method
+        prompt_embeds = prompt_embeds.repeat(1, num_images_per_prompt, 1)
+        prompt_embeds = prompt_embeds.view(batch_size * num_images_per_prompt, seq_len, -1)
+
+        return prompt_embeds
+
+    def _get_clip_prompt_embeds(
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/flux/pipeline_flux_kontext.py:311]: missing `#Copied from`

File: src/diffusers/pipelines/flux/pipeline_flux_kontext.py [hunk 4]
```diff
@@ -0,0 +1,1116 @@
...
+        prompt_embeds = prompt_embeds.pooler_output
+        prompt_embeds = prompt_embeds.to(dtype=self.text_encoder.dtype, device=device)
+
+        # duplicate text embeddings for each generation per prompt, using mps friendly method
+        prompt_embeds = prompt_embeds.repeat(1, num_images_per_prompt)
+        prompt_embeds = prompt_embeds.view(batch_size * num_images_per_prompt, -1)
+
+        return prompt_embeds
+
+    def encode_prompt(
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/flux/pipeline_flux_kontext.py:356]: `#Copied from`?

File: src/diffusers/pipelines/flux/pipeline_flux_kontext.py [hunk 5]
```diff
@@ -0,0 +1,1116 @@
...
+
+EXAMPLE_DOC_STRING = """
+    Examples:
+        ```py
+        >>> import torch
+        >>> from diffusers import FluxKontextPipeline
+        >>> from diffusers.utils import load_image
+
+        >>> pipe = FluxKontextPipeline.from_pretrained(
+        ...     "black-forest-labs/FLUX.1-kontext", transformer=transformer, torch_dtype=torch.bfloat16
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/flux/pipeline_flux_kontext.py]: same here, missing transformer too

**Review by yiyixuxu — APPROVED** [review]
awesome! looking good
just I think docstring example is broken
and make sure to add all the `#Copied from` - lets merge in quickly :)

File: docs/source/en/api/pipelines/flux.md (hunk 1 above)
Comment (a-r-r-o-w) [reply docs/source/en/api/pipelines/flux.md]: Thanks, nice catch! transformer part is not needed

Comment (sayakpaul) [reply docs/source/en/api/pipelines/flux.md]: This is the model ID: https://huggingface.co/black-forest-labs/FLUX.1-Kontext-dev

File: src/diffusers/pipelines/flux/pipeline_flux_kontext.py (hunk 3 above)
Comment (a-r-r-o-w) [reply src/diffusers/pipelines/flux/pipeline_flux_kontext.py:311]: done!
